samples/Ark.ReferenceProject/Core/Ark.Reference.Core.Tests/Features/BookPrintProcessSteps.cs

```diff
@@ -0,0 +1,207 @@
+using Ark.Reference.Core.Common.Dto;
+using Ark.Reference.Core.Common.Enum;
+using Ark.Tools.Core;
+
+using AwesomeAssertions;
+
+using Flurl;
+using Flurl.Http;
+
+using Reqnroll;
+
+using System;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+
+namespace Ark.Reference.Core.Tests.Features
+{
+    [Binding]
+    public sealed class BookPrintProcessSteps
+    {
+        private readonly TestClient _client;
+        private readonly string _controllerName = "bookprintprocess";
+        private int? _currentBookId;
+        private int? _currentPrintProcessId;
+        private HttpStatusCode? _lastStatusCode;
+        private string? _lastErrorContent;
+        private BookPrintProcess.V1.Output? _currentPrintProcess;
+
+        public BookPrintProcessSteps(TestClient client)
+        {
+            _client = client;
+        }
+
+        [Given(@"I have created a book with title ""(.*)"" and author ""(.*)""")]
+        public void GivenIHaveCreatedABookWithTitleAndAuthor(string title, string author)
+        {
+            var request = new Book.V1.Create
+            {
+                Title = title,
+                Author = author,
+                Genre = BookGenre.Fiction,
+                ISBN = "978-0-123456-78-9"
+            };
+
+            _client.PostAsJson("book", request);
+            _client.ThenTheRequestSucceded();
+            
+            var result = _client.ReadAs<Book.V1.Output>();
+            _currentBookId = result.Id;
+        }
+
+        [Given(@"I have created a book print process for that book")]
+        [When(@"I create a book print process for that book")]
+        public void WhenICreateABookPrintProcessForThatBook()
+        {
+            var request = new BookPrintProcess.V1.Create
+            {
+                BookId = _currentBookId!.Value,
+                ShouldFail = false
+            };
+
+            try
+            {
+                _client.PostAsJson(_controllerName, request);
+                
+                if (_client.LastStatusCodeIsSuccess())
+                {
+                    _currentPrintProcess = _client.ReadAs<BookPrintProcess.V1.Output>();
+                    _currentPrintProcessId = _currentPrintProcess!.BookPrintProcessId;
+                    _lastStatusCode = _client.GetLastStatusCode();
+                }
+                else
+                {
+                    _lastStatusCode = _client.GetLastStatusCode();
+                    _lastErrorContent = _client.ReadAsString();
```

Comment:
don't read errors this way. parse as generic problem details, if you expect so.

Client should already have some helpers